app/questionnaire/group_dependencies.py

```diff
@@ -64,15 +64,41 @@ def get_group_dependencies(schema):
         if group.get('routing_rules') and not _group_has_relationship_type(schema, group['blocks']):
             for routing_rule in group['routing_rules']:
                 if _routing_rule_has_repeat(routing_rule):
-                    if 'group_ids' in routing_rule['repeat']:
-                        dependencies.update(
-                            _get_dependency_drivers_from_group_ids(group['id'],
-                                                                   routing_rule['repeat']['group_ids']))
-                    else:
-                        dependencies.update(
-                            _get_dependency_driver_from_answer_id(schema,
-                                                                  group['id'],
-                                                                  routing_rule['repeat']['answer_id']))
+                    dependencies.update(_build_routing_dependencies(schema, routing_rule, group['id']))
+
+                    # Only when the group is a repeat
```

Comment:
I don't think the comment is helpful as there isn't a conditional around the next line.  Either expand this comment to give more information or remove it.